jupyterhub/schema.yaml

```diff
@@ -308,20 +308,38 @@ properties:
         description: |
           Extra environment variables that should be set for the hub pod.
 
+          Environment variables are usually used to:
+            - Pass parameters to some custom code in `hub.extraConfig`.
+            - Configure code running in the hub pod, such as an authenticator or
+              spawner.
+
           ```yaml
           hub:
             extraEnv:
-              MY_ENV_VARS_NAME: "my env vars value"
+              # short notation, only for hardcoded string values
+              MY_ENV_VARS_NAME1: "my env var value 1"
+
+              # long notation, Kubernetes native syntax
+              dummyKey1:
+                name: HUB_NAMESPACE
+                valueFrom:
+                  fieldRef:
+                    fieldPath: metadata.namespace
+              dummyKey2:
+                name: PREFIXED_HUB_NAMESPACE
+                value: "my-prefix-$(HUB_NAMESPACE)"
+              dummyKey3:
+                name: SECRET_VALUE
+                valueFrom:
+                  secretKeyRef:
+                    name: my-k8s-secret
+                    key: password
```

Comment:
I can see why you've done it this way, but my first instinct is it's confusing. Environment variables are already key-value pairs so adding the extra dummy keys seems unnecessary. Is there ever a case where the `name` key would not exist? If not what do you think of:
```yaml
              PREFIXED_HUB_NAMESPACE:
                value: "my-prefix-$(HUB_NAMESPACE)"
              SECRET_VALUE:
                valueFrom:
                  secretKeyRef:
                    name: my-k8s-secret
                    key: password
```
It doesn't match the k8s spec for an env-var, but neither does requiring a dict of env-vars instead of a list.

Reply:
I considered this and felt that the most important principle to follow was to stick with valid k8s specifications. A compromise could be to accept leaving out the `name: value` field and automatically populating it using the top level key name and let the template rendering fail if a `name: something` field was specified that mismatched the top level key name.

What do you think about that approach? I want to ensure there isn't any confusion about what takes priority etc down the line.

Reply:
Syntax updated. The `name: something` will take precedence and the top level key name is a fallback now.